x-pack/plugins/ml/public/application/components/controls/select_severity/select_severity.test.tsx

```diff
@@ -4,96 +4,63 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
-import React from 'react';
-import { act } from 'react-dom/test-utils';
-import { MemoryRouter } from 'react-router-dom';
-import { mount } from 'enzyme';
-
-import { EuiSuperSelect } from '@elastic/eui';
-
-import { UrlStateProvider } from '@kbn/ml-url-state';
-
-import { SelectSeverity } from './select_severity';
+import React, { useState } from 'react';
+import { render, act, fireEvent, waitFor } from '@testing-library/react';
+import { __IntlProvider as IntlProvider } from '@kbn/i18n-react';
+
+import { SelectSeverity, SEVERITY_OPTIONS } from './select_severity';
+
+// The following mock setup is necessary so that we can simulate
+// both triggering the update callback and the internal state update
+// to update the dropdown to the new state.
+const mockSeverityOptions = SEVERITY_OPTIONS;
+const mockUpdateCallback = jest.fn();
+const mockUseState = jest.fn().mockImplementation(useState);
+jest.mock('@kbn/ml-url-state', () => ({
+  usePageUrlState: () => {
+    const [severity, setSeverity] = mockUseState(mockSeverityOptions[0]);
+    return [severity, mockUpdateCallback.mockImplementation((d) => setSeverity(d))];
+  },
+}));
 
 describe('SelectSeverity', () => {
-  test('creates correct severity options and initial selected value', () => {
-    const wrapper = mount(
-      <MemoryRouter>
-        <UrlStateProvider>
-          <SelectSeverity />
-        </UrlStateProvider>
-      </MemoryRouter>
-    );
-    const select = wrapper.find(EuiSuperSelect);
-
-    const options = select.props().options;
-    const defaultSelectedValue = select.props().valueOfSelected;
-
-    expect(defaultSelectedValue).toBe('warning');
-    expect(options.length).toEqual(4);
-
-    // excpect options Array to equal Array containing Object that contains the property
-    expect(options).toEqual(
-      expect.arrayContaining([
-        expect.objectContaining({
-          value: 'warning',
-        }),
-      ])
-    );
-
-    expect(options).toEqual(
-      expect.arrayContaining([
-        expect.objectContaining({
-          value: 'minor',
-        }),
-      ])
-    );
-
-    expect(options).toEqual(
-      expect.arrayContaining([
-        expect.objectContaining({
-          value: 'major',
-        }),
-      ])
+  it('updates the severity option correctly on click', async () => {
```

Comment:
I reckon we need to clear mocks before or after each test 
```suggestion
  afterEach(() => {
    jest.clearAllMocks();
  });
  
  it('updates the severity option correctly on click', async () => {
```

Reply:
It's just a single test per file for now but doesn't hurt to add it, so updated in bd9a439.